openapi/components/schemas/Quote/CommonQuote.yaml

```
@@ -0,0 +1,137 @@
+      - accepted
+      - rejected
+      - cancelled
+      - expired
+    x-enumDescriptions:
+      draft: Editing is allowed, can not be sent to the customer.
+      issued: Editing is not allowed, can be sent to customer.
+      accepted: Customer accepted the quote and created an order.
+      rejected: Customer rejected the quote.
+      cancelled: Organization cancelled the quote.
+      expired: Expired before customer or organization interaction.
+  websiteId:
+    description: The website identifier string.
+    allOf:
+      - $ref: ../ResourceId.yaml
+  customerId:
+    $ref: ../CustomerId.yaml
+  orderId:
+    description: The order identifier string.
+    readOnly: true
+    nullable: true
+    allOf: 
+      - $ref: ../ResourceId.yaml
+  items:
+    type: array
+    minItems: 1
+    items:
+      $ref: ../Subscription/OrderItem.yaml
+  deliveryAddress:
+    description: Order delivery address.
+    nullable: true
+    allOf:
+      - $ref: ../Contact/ContactObject.yaml
+  billingAddress:
+    description: Order billing address.
+    nullable: true
+    allOf:
+      - $ref: ../Contact/ContactObject.yaml
+  invoicePreview:
+    type: object
+    description: The preview of initial quote invoice.
+    readOnly: true
+    properties:
+      currency:
+        description: Invoice currency.
+        allOf:
+          - $ref: ../../schemas/CurrencyCode.yaml
+      amount:
+        description: Amount of the invoice.
+        type: number
+        x-type: Money
+        format: double
+      amountDue:
+        description: Amount that is due on the invoice.
+        type: number
+        x-type: Money
+        format: double
+      subtotalAmount:
+        description: Subtotal amount of the invoice.
+        type: number
+        x-type: Money
+        format: double
+      discountAmount:
+        description: Discount amount that is applied to the invoice.
+        type: number
+        x-type: Money
+        format: double
+      tax:
+        $ref: ../Invoices/InvoiceTax.yaml
+      items:
+        type: array
+        description: Invoice items array.
+        readOnly: true
+        items:
+          $ref: ../Invoices/InvoiceItem.yaml
+      discounts:
+        type: array
+        description: Discounts applied.
+        readOnly: true
+        items:
+          allOf:
+            - $ref: ../Coupon/InvoiceDiscount.yaml
+  shipping:
+    $ref: ../Shipping.yaml
+  notes:
+    description: Notes for the customer which will be displayed on the quote.
+    type: string
+  expirationTime:
+    description: Expiration date-time. Defaults to one month.
+    type: string
+    format: date-time
+  issuedTime:
+    description: Date and time at which the quote was issued.
+    nullable: true
+    readOnly: true
+    allOf:
+      - $ref: ../ServerTimestamp.yaml
+  acceptedTime:
+    description: Date and time at which the quote was accepted.
+    nullable: true
+    readOnly: true
+    allOf:
+      - $ref: ../ServerTimestamp.yaml
```

This type of schema definition ends up being invalid because the `nullable` conflicts with the `allOf`. Using server timestamp is wrong in almost all cases. This isn't really acceptable reuse. Besides being incorrect, it introduces obscurity and doesn't even save lines.
```suggestion
  acceptedTime:
    description: Date and time at which the quote was accepted.
    nullable: true
    readOnly: true
    type: string
    format: date-time
```